swap_meet/vendor.py

```diff
@@ -0,0 +1,207 @@
+    #returns list of class Item's objects inventory = [Item_a, Item_b, Item_c]
+    
+    def get_by_category(self, category):
+        new_list_with_items = []
+        
+        #iterate on the list for each category (state) and check if the item is in there 
+        for thing in self.inventory:
+            if thing.category == category:
+                new_list_with_items.append(thing)
+    
+        return new_list_with_items
+
+#_______________Wave 3___________________________
+   
+    #helper function
+    def does_item_exist(self, item, item_list):
+        for i in item_list:
+            if i == item:
+                return True
+        return False
+            
+    def swap_items(self, Vendor, my_item, their_item):
+        my_inventory = self.inventory
+        their_inventory = Vendor.inventory
+        
+        if self.does_item_exist(my_item, my_inventory) == False or self.does_item_exist(their_item, their_inventory) == False:
+            return False
+
+        # at this point, both vendors have item available to swap
+        my_inventory.remove(my_item)
+        their_inventory.append(my_item)
+        
+        their_inventory.remove(their_item)
+        my_inventory.append(their_item)
+        
+        return True
+            
+#_______________wave 4___________________________
+
+    def swap_first_item(self, Vendor):
+        my_inventory = self.inventory
+        their_inventory = Vendor.inventory
+        
+        #return flase if either of the 2 vendors have an empty list
+        if (len(my_inventory) == 0) or (len(their_inventory) == 0):
+           return False
+        
+        #remove first item from both vendors and add it to the other's list
+        my_inventory.append(their_inventory.pop(0)) 
+        their_inventory.append(my_inventory.pop(0))     
+        
+        return True
+        
+#_______________wave 6___________________________
+
+    def get_best_by_category(self, category):
+        
+        best_item = None
+        
+        # step 1: Get list of items with the desired category
+        new_list_by_category = self.get_by_category(category)
+               
+        # If there are no items in the inventory that match the category, it returns None       
+        if len(new_list_by_category) == 0:
+            return None       
+        
+        # step 2: Get the best item (with highest condition) from the list in Step1, if there are matching items category
+        list_by_condition = []
+        for thing in new_list_by_category:
+            list_by_condition.append(thing.condition)
+            
+        max_condition_value = max(list_by_condition) 
+        
+        #returns a single item even if there are duplicates with best condtition filtered by category
+        for thing in new_list_by_category:
+            if thing.condition == max_condition_value:
+                best_item = thing
+                break
```

Comment:
Notice that you have two of the same loops here. Consider how you could put all the logic in a single loop, and compare the value of the current item with the highest value as you go.